README.md

```diff
@@ -170,9 +170,56 @@ The port number for the HEC token or the HEC load balancer. The default value is
 
 Identifier for the HEC token.
 
-### fields (init) (optional)
+### metrics_from_event (bool) (optional)
 
-Lets you specify the index-time fields for the event data type, or metric dimensions for the metric data type. Null value fields are removed.
+When `data_type` is set to "metric", the ingest API will treat every key-value pair in the input event as a metric name-value pair. Set `metrics_from_event` to `false` to disable this behavior and use `metric_name_key` and `metric_value_key` to define metrics. The default value is `true`.
+
+### metric_name_key (string) (optional)
+
+Field name that contains the metric name. This parameter only works in conjunction with the `metrics_from_event` paramter. When this prameter is set, the `metrics_from_event` parameter is automatically set to `false`.
+
+### metric_value_key (string) (optional)
+
+Field name that contains the metric value, this parameter is required when `metric_name_key` is configured.
+
+### coerce_to_utf8 (bool) (optional)
+
+Indicates whether to allow non-UTF-8 characters in user logs. If set to `true`, any non-UTF-8 character is replaced by the string specified in `non_utf8_replacement_string`. If set to `false`, the Ingest API errors out any non-UTF-8 characters. This parameter is set to `true` by default.
+
+### non_utf8_replacement_string (string) (optional)
+
+If `coerce_to_utf8` is set to `true`, any non-UTF-8 character is replaced by the string you specify in this parameter. The parameter is set to `' '` by default.
+
+### Parameters for `splunk_ingest_api`
```

Comment:
Make the split of params for different "modes" clearer. Could use tables to separate like used in https://github.com/splunk/kafka-connect-splunk/blob/develop/README.md

Reply:
Will do for a future release.